guildwars2/commerce.py

```diff
@@ -14,90 +16,127 @@ class CommerceMixin:
+        for result in sells:
+            item_id_sells += str(result["item_id"]) + ","
+            if result["item_id"] not in dup_item_sells:
+                dup_item_sells[result["item_id"]] = len(dup_item_sells)
 
-        for result in results:
-            index = dup_item[result["item_id"]]
-            price = result["price"]
-            itemdoc = await self.fetch_item(result["item_id"])
-            quantity = result["quantity"]
-            item_name = itemdoc["name"]
-            offers = listings[index][state]
-            max_price = offers[0]["unit_price"]
-            undercuts = 0
-            op = operator.lt if state == "buys" else operator.gt
-            for offer in offers:
-                if op(offer["unit_price"], price):
-                    break
-                undercuts += offer["listings"]
-            undercuts = "· Undercuts: {}".format(
-                undercuts) if undercuts else ""
-            if quantity == 1:
-                total = ""
-            else:
-                total = " - Total: " + self.gold_to_coins(
-                    ctx, quantity * price)
-            data.add_field(
-                name=item_name,
-                value="{} x {}{}\nMax. offer: {} {}".format(
-                    quantity, self.gold_to_coins(ctx, price), total,
-                    self.gold_to_coins(ctx, max_price), undercuts),
-                inline=False)
+        # Get listings
+        listings_buys = await call_for_listing(item_id_buys)
+        listings_sells = await call_for_listing(item_id_sells)
+
+        # Generate strings
+        sells_string = await generate_listing_output(sells, dup_item_sells,
+                                                     "sells", listings_sells)
+        buys_string = await generate_listing_output(buys, dup_item_buys,
+                                                    "buys", listings_buys)
 
+        # Display inline if both are small enough
+        if sells_string and buys_string and len(sells_string) < 1024 and len(
+                buys_string) < 1024:
+            data = embed_list_lines(
+                data, buys_string, "> **SELLS**", inline=True)
+            data = embed_list_lines(
+                data, sells_string, "> **BUYS**", inline=True)
+        else:
+            if sells_string:
+                data = embed_list_lines(
+                    data, zero_width_space, "> **SELLS**", inline=False)
+                data = embed_list_lines(
+                    data, sells_string, zero_width_space, inline=True)
+            if buys_string:
+                data = embed_list_lines(
+                    data, zero_width_space, "> **BUYS**", inline=False)
+                data = embed_list_lines(
+                    data, buys_string, zero_width_space, inline=True)
         try:
             await ctx.send(embed=data)
         except discord.HTTPException:
             await ctx.send("Need permission to embed links")
```

Comment:
A lot of old, weird code in this command that needs to be rewritten. But the main thing I realized is that it doesn't work at all on mobile - buys and sells are mixed up. Hmm